static/src/javascripts/projects/common/modules/ui/objectfit-polyfill.js

```diff
@@ -0,0 +1,42 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/$'
+], function (
+    fastdomPromise,
+    $
+) {
+    // Polyfill for Object-fit: https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit
+
+    function getStyle (el) {
+        var style = getComputedStyle(el);
+        return style['object-fit'];
+    }
+
+    function setBackgroundImage(src) {
+        return fastdomPromise.write(function () {
+            $('.js-background-image').each(function () {
+                var $container = $(this);
+
+                if (src) {
+                    $container
+                        .css('backgroundImage', 'url(' + src + ')')
+                        .addClass('immersive-main-media--fallback');
+                }
+            });
+        });
+    }
+
+    function init() {
+        return fastdomPromise.read(function () {
+            return document.getElementsByClassName('js-immersive-main-media__img')[0];
```

Comment:
Thanks @jamesgorrie. Good catch! 

If it is a really old gallery it may not have any an immersive-main-media. If that is the case, this line returns `undefined`. I have added a check to make sure the picture is there before getting the style.